Hilt 에러 해결

### Describe

Hilt에러를 해결해야합니다.

### Additional

error log

```kotlin
/Users/yongjinlee/Desktop/android-project/shoes-of-world/app/build/generated/hilt/component_sources/debug/com/yongjincompany/shoesofworld/ShoesOfWorldApplication_HiltComponents.java:135: error: [Dagger/MissingBinding] com.yongjincompany.core.domain.repository.ShoesRepository cannot be provided without an @Provides-annotated method.
  public abstract static class SingletonC implements ShoesOfWorldApplication_GeneratedInjector,
                         ^
  
      com.yongjincompany.core.domain.repository.ShoesRepository is injected at
          [com.yongjincompany.shoesofworld.ShoesOfWorldApplication_HiltComponents.ViewModelC] com.yongjincompany.core.domain.usecase.FetchAllShoesUseCase(shoesRepository)
      com.yongjincompany.core.domain.usecase.FetchAllShoesUseCase is injected at
          [com.yongjincompany.shoesofworld.ShoesOfWorldApplication_HiltComponents.ViewModelC] com.yongjincompany.feature.home.HomeViewModel(fetchAllShoesUseCase)
      com.yongjincompany.feature.home.HomeViewModel is injected at
          [com.yongjincompany.shoesofworld.ShoesOfWorldApplication_HiltComponents.ViewModelC] com.yongjincompany.feature.home.HomeViewModel_HiltModules.BindsModule.binds(arg0)
      @dagger.hilt.android.internal.lifecycle.HiltViewModelMap java.util.Map<java.lang.Class<?>,javax.inject.Provider<androidx.lifecycle.ViewModel>> is requested at
          [com.yongjincompany.shoesofworld.ShoesOfWorldApplication_HiltComponents.ViewModelC] dagger.hilt.android.internal.lifecycle.HiltViewModelFactory.ViewModelFactoriesEntryPoint.getHiltViewModelMap() [com.yongjincompany.shoesofworld.ShoesOfWorldApplication_HiltComponents.SingletonC → com.yongjincompany.shoesofworld.ShoesOfWorldApplication_HiltComponents.ActivityRetainedC → com.yongjincompany.shoesofworld.ShoesOfWorldApplication_HiltComponents.ViewModelC]
```

--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -36,6 +36,7 @@ android {
 dependencies {
     implementation(projects.feature.home)
     implementation(projects.core.designsystem)
+    implementation(projects.core.data)
 
     implementation(libs.androidx.core.ktx)
     implementation(libs.androidx.appcompat)

--- a/core/data/src/main/java/com/yongjincompany/core/data/repository/ShoesRepositoryImpl.kt
+++ b/core/data/src/main/java/com/yongjincompany/core/data/repository/ShoesRepositoryImpl.kt
@@ -15,18 +15,17 @@ import javax.inject.Inject
 
 class ShoesRepositoryImpl @Inject constructor(
     private val remoteShoesDataSource: RemoteShoesDataSource,
-    private val localShoesDataSource: LocalShoesDataSource
+    private val localShoesDataSource: LocalShoesDataSource,
 ) : ShoesRepository {
-    override suspend fun fetchAllShoesList(): Flow<List<Shoes>> = flow {
-        emit(localShoesDataSource.selectAllFromShoesTable().map { it.localEntityToDomainEntity() })
-
+    override fun fetchAllShoesList(): Flow<List<Shoes>> = flow {
         runCatching {
             remoteShoesDataSource.fetchAllShoesList()
         }.onSuccess { allShoesList ->
             localShoesDataSource.insertShoes(allShoesList.map { shoes -> shoes.responseToLocalEntity() })
             emit(allShoesList.map { it.responseToEntity() })
         }.onFailure {
-            //TODO: errorHandling구현하면서 처리 필요.
+            emit(localShoesDataSource.selectAllFromShoesTable().map { it.localEntityToDomainEntity() })
         }
-    }.flowOn(Dispatchers.IO)
+    }
+        .flowOn(Dispatchers.IO)
 }

--- a/core/domain/src/main/java/com/yongjincompany/core/domain/repository/ShoesRepository.kt
+++ b/core/domain/src/main/java/com/yongjincompany/core/domain/repository/ShoesRepository.kt
@@ -4,5 +4,5 @@ import com.yongjincompany.core.domain.entity.Shoes
 import kotlinx.coroutines.flow.Flow
 
 interface ShoesRepository {
-    suspend fun fetchAllShoesList(): Flow<List<Shoes>>
+    fun fetchAllShoesList(): Flow<List<Shoes>>
 }

